// @patternfly/react-core/dist/umd/components/Tabs/Tab.js
"use strict";

(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react", "prop-types"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react"), require("prop-types"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.react, global.propTypes);
    global.undefined = mod.exports;
  }
})(void 0, function (exports, _react, _propTypes) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};

    var target = _objectWithoutPropertiesLoose(source, excluded);

    var key, i;

    if (Object.getOwnPropertySymbols) {
      var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

      for (i = 0; i < sourceSymbolKeys.length; i++) {
        key = sourceSymbolKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
        target[key] = source[key];
      }
    }

    return target;
  }

  function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;

    for (i = 0; i < sourceKeys.length; i++) {
      key = sourceKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      target[key] = source[key];
    }

    return target;
  }

  var propTypes = {
    /** content rendered inside the Tab content area. */
    children: _propTypes2.default.node,

    /** additional classes added to the Modal */
    className: _propTypes2.default.string,

    /** Tab title */
    title: _propTypes2.default.string.isRequired,

    /** uniquely identifies the tab */
    eventKey: _propTypes2.default.number.isRequired
  };
  var defaultProps = {
    children: null,
    className: ''
  };

  var Tab = function Tab(_ref) {
    var className = _ref.className,
        children = _ref.children,
        title = _ref.title,
        eventKey = _ref.eventKey,
        props = _objectWithoutProperties(_ref, ["className", "children", "title", "eventKey"]);

    return _react2.default.createElement(_react2.default.Fragment, null, children);
  };

  Tab.propTypes = propTypes;
  Tab.defaultProps = defaultProps;
  exports.default = Tab;
});
