// @patternfly/react-core/dist/js/components/Text/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "TextContent", {
  enumerable: true,
  get: function get() {
    return _TextContent.default;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _Text.default;
  }
});
Object.defineProperty(exports, "TextVariants", {
  enumerable: true,
  get: function get() {
    return _Text.TextVariants;
  }
});
Object.defineProperty(exports, "TextList", {
  enumerable: true,
  get: function get() {
    return _TextList.default;
  }
});
Object.defineProperty(exports, "TextListVariants", {
  enumerable: true,
  get: function get() {
    return _TextList.TextListVariants;
  }
});
Object.defineProperty(exports, "TextListItem", {
  enumerable: true,
  get: function get() {
    return _TextListItem.default;
  }
});
Object.defineProperty(exports, "TextListItemVariants", {
  enumerable: true,
  get: function get() {
    return _TextListItem.TextListItemVariants;
  }
});

var _TextContent = _interopRequireDefault(require("./TextContent"));

var _Text = _interopRequireWildcard(require("./Text"));

var _TextList = _interopRequireWildcard(require("./TextList"));

var _TextListItem = _interopRequireWildcard(require("./TextListItem"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
