// @patternfly/react-core/dist/esm/components/Select/SingleSelect.js
function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import styles from "../../@patternfly/patternfly/components/Select/select.css.js";
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import { keyHandler } from '../../helpers/util';
var propTypes = {
  /** Content rendered inside the SingleSelect */
  children: PropTypes.node.isRequired,

  /** Additional classes added to the SingleSelect control */
  className: PropTypes.string,

  /** Flag indicating the Select is expanded */
  isExpanded: PropTypes.bool,

  /** Internal flag indicating whether select was opened via keyboard */
  openedOnEnter: PropTypes.bool,

  /** Currently selected option */
  selected: PropTypes.string,

  /** Additional props are spread to the container <select> */
  '': PropTypes.any
};
var defaultProps = {
  className: '',
  isExpanded: false,
  openedOnEnter: false,
  selected: ''
};

var SingleSelect =
/*#__PURE__*/
function (_React$Component) {
  _inherits(SingleSelect, _React$Component);

  function SingleSelect() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, SingleSelect);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(SingleSelect)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "refCollection", []);

    _defineProperty(_assertThisInitialized(_this), "sendRef", function (ref, index) {
      _this.refCollection[index] = ref;
    });

    _defineProperty(_assertThisInitialized(_this), "childKeyHandler", function (index, position) {
      keyHandler(index, position, _this.refCollection, _this.props.children);
    });

    return _this;
  }

  _createClass(SingleSelect, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (this.props.openedOnEnter) {
        var selectedRef = this.refCollection.filter(function (ref) {
          return ref.classList.contains('pf-c-select__menu-item--match');
        });
        selectedRef && selectedRef[0] ? selectedRef[0].focus() : this.refCollection[0].focus();
      }
    }
  }, {
    key: "extendChildren",
    value: function extendChildren() {
      var _this2 = this;

      return React.Children.map(this.props.children, function (child, index) {
        return React.cloneElement(child, {
          isSelected: _this2.props.selected === child.props.value,
          sendRef: _this2.sendRef,
          keyHandler: _this2.childKeyHandler,
          index: index
        });
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          className = _this$props.className,
          isExpanded = _this$props.isExpanded,
          openedOnEnter = _this$props.openedOnEnter,
          selected = _this$props.selected,
          props = _objectWithoutProperties(_this$props, ["children", "className", "isExpanded", "openedOnEnter", "selected"]);

      this.renderedChildren = this.extendChildren();
      return React.createElement("ul", _extends({}, props, {
        className: css(styles.selectMenu, className),
        role: "listbox"
      }), this.renderedChildren);
    }
  }]);

  return SingleSelect;
}(React.Component);

SingleSelect.propTypes = propTypes;
SingleSelect.defaultProps = defaultProps;
export default SingleSelect;
